src/reference-designs/tkg-edge.md

```diff
@@ -0,0 +1,885 @@
+
+##### Service Engine Configuration
+
+| Service Engine Configuration | Minimal Specification |
+| --- | --- |
+| vCPUs | 1   |
+| Memory (GB) | 1   |
+| Hard disk (GB) | 15  |
+| Number of connected network adapters | 2   |
+| Number of service engines per Service Engine Group | 2   |
+
+See [Sizing Service Engines][410] for guidance on sizing your SEs. In summary, for Intel v4 CPUs, for example Intel® Xeon® CPU E5-2686 v4 @ 2.30GHz:
+
+| Performance metric | 1 vCPU core |
+| --- | --- |
+| Throughput | 4 Gb/s |
+| Connections/s | 40k |
+| SSL Throughput | 1 Gb/s |
+| SSL TPS (RSA2K) | ~600 |
+| SSL TPS (ECC) | 2500 |
+
+Multiple performance vectors or features may have an impact on performance. For instance, to achieve 1 Gb/s of SSL throughput and 2000 TPS of SSL with EC certificates, Avi recommends two cores.
+
+Avi Service Engines may be configured with as little as 1 vCPU core and 1 GB RAM, or up to 36 vCPU cores and 128 GB RAM.
+
+### Contour for Kubernetes Ingress Routing
+
+Tanzu Editions also includes [Contour](https://projectcontour.io/) for Kubernetes Ingress routing. Contour and can be installed on any Tanzu Kubernetes Grid cluster with ease. For more information, see [Implementing Ingress Control with Contour](https://docs.vmware.com/en/VMware-Tanzu-Kubernetes-Grid/1.4/vmware-tanzu-kubernetes-grid-14/GUID-packages-ingress-contour.html). Contour provides **Layer 7 based routing** to Kubernetes services.
+
+## Container Registry
+
+### Main Datacenter
+
+[Harbor Container Registry](https://goharbor.io/) can be deployed to both the centralized datacenter as well as to each edge location.  Using Harbor, replications can be configured such that all container images which get pushed into the centralized harbor are automatically replicated to each edge. Each edge deployment can reference its local harbor for efficient network resource utilization.
+
+![](./img/tkg-edge/tkg-edge-harbor.drawio.png)
+
+Figure: Harbor registry federated topology
+
+[Replications](https://goharbor.io/docs/2.0.0/administration/configuring-replication/create-replication-rules/) can be defined as manual, scheduled, or event-based.  Manual or scheduled are best suited for initial bulk syncs, where-as event-based can be used to replicate changes on-going.
+
+Schedule replication during low activity (outside of working hours)
+
+![](./img/tkg-edge/f312de8ea37cad334af63d608a37f3d6.png)
+
+Figure: Harbor edge site replications
+
+![](./img/tkg-edge/0547aae877598d9975f066451cb5b268.png)
+Figure: Harbor registry webhooks for edge site continuous delivery
+
+### At the Edge
+
+| Decision ID | Design Decision | Design Justification | Design Implication |
+| --- | --- | --- | --- |
+| `TZEDG-EDGE-REG-01` | Use Pull | more resilient, since the edge Harbors can pull in their own time, and it keeps configuration simple | configuration needs to be done at the edge site which might implies |
+
+Table: Design Decisions
```

Comment:
Move to line 432. Write as a lead, for example: The following table describes the justification and implication of the design decisions in the architecture.